include/rocksdb/memtablerep.h

```diff
@@ -120,6 +120,20 @@ class MemTableRep {
     return true;
   }
 
+  // Same as ::InsertWithHint, but allow concurrnet write
```

Comment:
We need to document the ownership of hint. In my understanding, the caller will own the object hint, correct?

Or, if that is the case, an even better idea is to change the argument to `std::unique_ptr<void>*`, so that is ownership is clear.